pilot/pkg/networking/core/cluster_traffic_policy.go

```diff
@@ -472,6 +473,19 @@ func (cb *ClusterBuilder) applyUpstreamProxyProtocol(
 		return
 	}
 	c := opts.mutable
+
+	// No existing transport; wrap RawBuffer.
+	if c.cluster.TransportSocket == nil && len(c.cluster.TransportSocketMatches) == 0 {
```

Comment:
How could this happend, either TransportSocket or TransportSocketMatches is set before calling this function

Reply:
Not if tls is disabled, then there is neither

Reply:
There is a unit test case covering this

Reply:
and the integration test also covers this